src/serde_util.rs

```diff
@@ -191,3 +194,37 @@ where
     }
     Ok(Some(vec))
 }
+
+pub(crate) fn serialize_indexed_map<S: Serializer, T: Serialize>(
+    map: &Option<HashMap<usize, T>>,
+    serializer: S,
+) -> std::result::Result<S::Ok, S::Error> {
+    if let Some(map) = map {
+        let string_map: BTreeMap<_, _> = map
+            .iter()
+            .map(|(index, result)| (index.to_string(), result))
+            .collect();
+        string_map.serialize(serializer)
+    } else {
+        serializer.serialize_none()
+    }
+}
+
+#[cfg(test)]
+pub(crate) fn deserialize_indexed_map<'de, D, T>(
+    deserializer: D,
+) -> std::result::Result<Option<HashMap<usize, T>>, D::Error>
+where
+    D: Deserializer<'de>,
+    T: serde::de::DeserializeOwned,
+{
+    use std::str::FromStr;
+
+    let string_map: HashMap<String, T> = HashMap::deserialize(deserializer)?;
+    Ok(Some(
+        string_map
+            .into_iter()
+            .map(|(index, t)| (usize::from_str(&index).unwrap(), t))
```

Comment:
Can the string parse error be translated to a deserialization error rather than unwrapped?

Reply:
yep, updated